File: TaskManeger/app/src/main/java/com/setbitzero/taskmaneger/activities/MainActivity.java
package com.setbitzero.taskmaneger.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.navigation.Navigation;

import android.annotation.SuppressLint;
import android.app.DatePickerDialog;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;

import com.google.android.material.button.MaterialButton;
import com.setbitzero.taskmaneger.R;
import com.setbitzero.taskmaneger.databinding.ActivityMainBinding;

import java.util.Calendar;

public class MainActivity extends AppCompatActivity{
    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        navigate();
        binding.addTask.setOnClickListener(v ->{
//                Navigation.findNavController(MainActivity.this, R.id.nav_controller).navigate(R.id.addTaskFragment);
//                binding.navigationBar.setVisibility(View.GONE);
//                binding.cardView.setVisibility(View.GONE);
            startActivity(new Intent(MainActivity.this, AddTaskActivity.class));
        });

        binding.searchView.setOnSearchClickListener(v -> {
            binding.spinner.setVisibility(View.GONE);
            binding.searchView.setVisibility(View.VISIBLE);
        });

        binding.searchView.setOnCloseListener(() -> {
            binding.spinner.setVisibility(View.VISIBLE);
            return false;
        });

    }

    @SuppressLint("NonConstantResourceId")
    private void navigate() {
        binding.navigationBar.setOnItemSelectedListener(item -> {

            switch (item.getItemId()){
                case R.id.allTask:
                    Navigation.findNavController(MainActivity.this, R.id.nav_controller).navigate(R.id.allListFragment);
                    break;

                case R.id.runningTask:
                    Navigation.findNavController(MainActivity.this, R.id.nav_controller).navigate(R.id.runningFragment2);
                    break;

                case R.id.pendingTask:
                    Navigation.findNavController(MainActivity.this, R.id.nav_controller).navigate(R.id.pendingFragment2);
                    break;

                case R.id.completeTask:
                    Navigation.findNavController(MainActivity.this, R.id.nav_controller).navigate(R.id.completeFragment);
                    break;

                case R.id.notification:
                    Navigation.findNavController(MainActivity.this, R.id.nav_controller).navigate(R.id.notificationFragment);
                    break;

                default:
            }

            return true;
        });
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        binding.navigationBar.setVisibility(View.VISIBLE);
        binding.topBar.setVisibility(View.VISIBLE);
    }

}

File: DMovies/app/src/main/java/com/setbitzero/dmovies/MainActivity.kt
package com.setbitzero.dmovies

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.navigation.findNavController
import com.setbitzero.dmovies.databinding.ActivityMainBinding
import com.setbitzero.dmovies.ui.search.SearchResultFragment


class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.searchItem.setOnClickListener{
            binding.searchItem.visibility = View.GONE
            binding.itemNotification.visibility = View.GONE
            binding.profileImage.visibility = View.GONE
            binding.searchMovieInL.visibility = View.VISIBLE
        }

        binding.searchMovieInL.setEndIconOnClickListener {
            binding.searchMovieInL.visibility = View.GONE
            binding.searchItem.visibility=View.VISIBLE
            binding.itemNotification.visibility = View.VISIBLE
            binding.profileImage.visibility = View.VISIBLE

            val query = binding.searchMovie.text.toString()
            sendQuery(query)
        }
    }

    private fun sendQuery(query:String){
        val args = SearchResultFragment()
        val transaction = supportFragmentManager.beginTransaction()
        val bundle = Bundle()
        bundle.putString("query", query)
        args.arguments = bundle
        transaction.replace(R.id.main_activity_container, args).commit()
        findNavController(R.id.nav_controller).navigate(R.id.navigation_search)
    }

}